modules/light-clients/08-wasm/keeper/keeper.go

```diff
@@ -44,10 +43,6 @@ func NewKeeperWithVM(
 		panic(errors.New("wasm VM must be not nil"))
 	}
 
-	if types.WasmVM != nil && !reflect.DeepEqual(types.WasmVM, vm) {
-		panic(errors.New("global Wasm VM instance should not be set to a different instance"))
-	}
```

Comment:
this check doesn't work with mock vm when you have other testing also uses the wasmvm, so I decided to remove it, especially since we are going to manage global assignment internally

Reply:
TBH I like this quite a bit, even though we're still dealing with global variables, removing the scope at which we assign them can never be a bad thing.